### PR TITLE
FF93 First changes for digest SHA256 support

### PR DESCRIPTION
**Pull request by hamishwillee**
FF93 supports SHA256 for digest authentication.  Docs work can be tracked in #8682

This is just the first part of the work. What I have discovered is that the HTTP Authorization stuff is overly focussed on basic authorization. This first fix is primarily to `WWW-Authenticate` to make it clear that this is part of a generic framework that may be modified for specific authorization schemes. 
Essentially with this change it still really only covers "Basic" but will be easy to extend for Digest when I do that (next).
If you're OK with that change I'll roll it out to the `Authorization` header. 

I will be adding more about schemas and what is supported but that is mostly going to be in one place - the HTTP authentication header OR BCD (in discussion). What I'd like to do is have in BCD along with the associated specs, but still in discussion. 

Following all this, I hope to actually be able to go in and add Digest information - and it will then be in a structure that anyone else can extend if they like.

### CODE REVIEW

**Comment by hamishwillee** [comment]
@teoli2003 Ready for review - mostly I want to sanity check `WWW-Authenticate` changes before rolling out to the rest. Idea is to make an extensible framework for the docs that anyone can build on, much like the design of the framework itself.

**Comment by gstrauss** [comment]
Suggestion:

>   - : See {{rfc(7616)}}. Firefox 93 and later support SHA-256 encryption. Previous versions only support MD5 hashing (not recommended).

"Firefox 93 and later support SHA-256 encryption if a SHA-256 challenge is offered by the server.  Previous versions only support MD5 hashing (not recommended)."

Somewhere close by, maybe in the paragraph above:  "HTTP over TLS is recommended for all HTTP Authentication."

**Comment by hamishwillee** [comment]
> Suggestion:

>>  : See {{rfc(7616)}}. Firefox 93 and later support SHA-256 encryption. Previous versions only support MD5 hashing (not recommended).

> "Firefox 93 and later support SHA-256 encryption if a SHA-256 challenge is offered by the server. Previous versions only support MD5 hashing (not recommended)."

> Somewhere close by, maybe in the paragraph above: "HTTP over TLS is recommended for all HTTP Authentication."

No. The FF93 supports SHA-256 irrespective of whether the server sends the challenge :-). I do understand what you mean, but I still wouldn't add this clarification because it is "default behaviour" that the challenge defines the algorithm. 

I'm coming back to this on Friday, but generally we need a better way than this to indicate support across browsers, and possibly also with popular servers - Apache etc. I'm kind of hoping we can use BCD for this, but probably worth building out as a table here in the first instance.

File: files/en-us/web/http/headers/www-authenticate/index.md [hunk 1]
```diff
@@ -11,9 +11,20 @@ browser-compat: http.headers.WWW-Authenticate
...
 {{HTTPSidebar}}
 
-The HTTP **`WWW-Authenticate`** response header defines the authentication method that should be used to gain access to a resource.
+The HTTP **`WWW-Authenticate`** response header defines the [HTTP authentication](/en-US/docs/Web/HTTP/Authentication) methods ("challenges") that might be used to gain access to a specific resource.
 
-The `WWW-Authenticate` header is sent along with a {{HTTPStatus("401")}} `Unauthorized` response.
+> **Note:** This header is part of the [General HTTP authentication framework](/en-US/docs/Web/HTTP/Authentication#the_general_http_authentication_framework), which can be used with a number of [authentication schemes](/en-US/docs/Web/HTTP/Authentication#authentication_schemes) .
+> Each "challenge" lists a scheme supported by the server and additional parameters that are defined for that scheme type. 
+
+A server using [HTTP authentication](/en-US/docs/Web/HTTP/Authentication) will respond with {{HTTPStatus("401")}} `Unauthorized` response to a request for a protected resource.
```
Comment (sideshowbarker) [inline files/en-us/web/http/headers/www-authenticate/index.md:19]:
```suggestion
A server using [HTTP authentication](/en-US/docs/Web/HTTP/Authentication) will respond with a {{HTTPStatus("401")}} `Unauthorized` response to a request for a protected resource.
```

File: files/en-us/web/http/headers/www-authenticate/index.md [hunk 2]
```diff
@@ -11,9 +11,20 @@ browser-compat: http.headers.WWW-Authenticate
...
+This response must include at least one `WWW-Authenticate` header and at least one {{Glossary("challenge")}}, to indicate what authentication schemes can be used to access the resource (and any additional data that each particular scheme needs).
+
+Multiple challenges are allowed in the one `WWW-Authenticate` header, and multiple `WWW-Authenticate` headers are allowed in the one response.
```
Comment (sideshowbarker) [inline files/en-us/web/http/headers/www-authenticate/index.md:22]:
```suggestion
Multiple challenges are allowed in one `WWW-Authenticate` header, and multiple `WWW-Authenticate` headers are allowed in one response.
```
…or ignore this suggestion if you in fact have *the* in there intentionally (which I can see is possible)

File: files/en-us/web/http/headers/www-authenticate/index.md [hunk 3]
```diff
@@ -11,9 +11,20 @@ browser-compat: http.headers.WWW-Authenticate
...
+A server may also include the `WWW-Authenticate` header in other response messages to indicate that supplying credentials might affect the response.
+
+After recieving `WWW-Authenticate` a client will typically prompt the user for credentials, and then re-request the resource.
```
Comment (sideshowbarker) [inline files/en-us/web/http/headers/www-authenticate/index.md:25]:
```suggestion
After receiving the `WWW-Authenticate` header, a client will typically prompt the user for credentials, and then re-request the resource.
```

File: files/en-us/web/http/headers/www-authenticate/index.md [hunk 4]
```diff
@@ -30,27 +41,84 @@ The `WWW-Authenticate` header is sent along with a {{HTTPStatus("401")}} `Unauth
 
 ## Syntax
 
+At least one challenge must be specified.
+Multiple challenges may be specified commas-separated in a single header, or in individual headers:
```
Comment (sideshowbarker) [inline files/en-us/web/http/headers/www-authenticate/index.md:45]:
```suggestion
Multiple challenges may be specified, comma-separated, in a single header, or in individual headers:
```

File: files/en-us/web/http/headers/www-authenticate/index.md [hunk 5]
```diff
@@ -30,27 +41,84 @@ The `WWW-Authenticate` header is sent along with a {{HTTPStatus("401")}} `Unauth
...
+These are listed below for some schemes.
+For others you may need to check the specifications:
+
+### Basic
+
+- **`<realm>`** {{optional_inline}}
+  - : As above.
+- **`charset="UTF-8"`** {{optional_inline}}
+  - : Tells the client the server's preferred encoding scheme when submitting a username and password.
+      The only allowed value is the case insensitive string "UTF-8".
```
Comment (sideshowbarker) [inline files/en-us/web/http/headers/www-authenticate/index.md:103]:
```suggestion
      The only allowed value is the case-insensitive string "UTF-8".
```

**Review by sideshowbarker — APPROVED** [review]

**Review by reschke — COMMENTED** [review]
Unless FF actually supports the "Digest" and "Want-Digest" fields (which I believe is not true), these pages probably should not be changed as part of the FF93 update.

**Comment by hamishwillee** [comment]
> Unless FF actually supports the "Digest" and "Want-Digest" fields (which I believe is not true), these pages probably should not be changed as part of the FF93 update.

@reschke Thanks. No need to worry.

The changes to the digest headers here are "cosmetic" - they were just minor general improvements - nothing to do with FF93. I made them before it became clear that the bugfix would not affect these headers and chose not to revert

**Pull request merged**